### PR TITLE
tauri-beta: init at 2.0.0-beta.17

### PR DESCRIPTION
**Pull request by Eveeifyeve**
## Description of changes

Adds a version of `cargo-tauri` for the beta version of tauri or known as v2 for people who want to use tauri v2

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Eveeifyeve** [comment]
I will fix the editor config tmr.

**Comment by Eveeifyeve** [comment]
SInce it is 12am

**Comment by NotAShelf** [comment]
nixpkgs does not use conventional commits. your commit message should be something like `tauri-beta: init at 2.0.0-beta.17`

**Comment by NotAShelf** [comment]
also, you'll probably want to put this in `pkgs/by-name` and not `pkgs/development`. but I don't really know the protocol for dev tools

File: pkgs/development/tools/rust/cargo-tauri/beta.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{ lib
```
Comment (AndersonTorres) [inline pkgs/development/tools/rust/cargo-tauri/beta.nix:1]: Move this to `pkgs/ca/cargoi-tauri/package.nix`.

P.S.: have you tested this package? How?

File: pkgs/development/tools/rust/cargo-tauri/beta.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+let
+  inherit (darwin.apple_sdk.frameworks) CoreServices Security SystemConfiguration;
+in
+rustPlatform.buildRustPackage rec {
+  pname = "tauri";
+  version = "2.0.0-beta.17";
+
+  src = fetchFromGitHub {
+    owner = "tauri-apps";
+    repo = pname;
```
Comment (AndersonTorres) [inline pkgs/development/tools/rust/cargo-tauri/beta.nix:24]:
```suggestion
    repo = "tauri";
```

**Comment by Eveeifyeve** [comment]
> nixpkgs does not use conventional commits. your commit message should be something like `tauri-beta: init at 2.0.0-beta.17`

Didn't know I know know.

**Comment by Eveeifyeve** [comment]
> also, you'll probably want to put this in `pkgs/by-name` and not `pkgs/development`. but I don't really know the protocol for dev tools

Sure but I kept it as the same thing since it adds a path to cargo so you can use `cargo tauri` as the previous job but just added a beta version.

File: pkgs/development/tools/rust/cargo-tauri/beta.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+  # Manually specify the sourceRoot since this crate depends on other crates in the workspace. Relevant info at
+  # https://discourse.nixos.org/t/difficulty-using-buildrustpackage-with-a-src-containing-multiple-cargo-workspaces/10202
+  sourceRoot = "${src.name}/tooling/cli";
+
+  cargoHash = "sha256-45R8oyWwI1rgclhC7jehZQrDEPanJPqLSpg0v6TimQo=";
+
+  buildInputs = [ openssl ] ++ lib.optionals stdenv.isLinux [ glibc libsoup cairo gtk3 webkitgtk ]
+    ++ lib.optionals stdenv.isDarwin [ CoreServices Security SystemConfiguration ];
+  nativeBuildInputs = [ pkg-config ];
```
Comment (AndersonTorres) [inline pkgs/development/tools/rust/cargo-tauri/beta.nix:37]:
```suggestion
  nativeBuildInputs = [ 
    pkg-config
  ];

  buildInputs = [ 
    openssl
  ] 
  ++ lib.optionals stdenv.isLinux [ 
    glibc libsoup cairo gtk3 webkitgtk 
  ]
  ++ lib.optionals stdenv.isDarwin [ 
    CoreServices 
    Security 
    SystemConfiguration 
  ];
  
  strictDeps = true;
```

File: pkgs/development/tools/rust/cargo-tauri/beta.nix [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+  meta = with lib; {
+    description = "Build smaller, faster, and more secure desktop applications with a web frontend";
+    mainProgram = "cargo-tauri-beta";
+    homepage = "https://tauri.app/";
+    license = with licenses; [ asl20 /* or */ mit ];
+    maintainers = with maintainers; [ dit7ya happysalada ];
+  };
+}
```
Comment (AndersonTorres) [inline pkgs/development/tools/rust/cargo-tauri/beta.nix:46]:
```suggestion
  meta = {
    description = "Build smaller, faster, and more secure desktop applications with a web frontend";
    homepage = "https://tauri.app/";
    license = with lib.licenses; [ asl20 /* or */ mit ];
    mainProgram = "cargo-tauri-beta";
    maintainers = with lib.maintainers; [ dit7ya happysalada ];
  };
}
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/rust/cargo-tauri/beta.nix (hunk 2 above)
Comment (Eveeifyeve) [reply pkgs/development/tools/rust/cargo-tauri/beta.nix:24]: The `pname is there since tauri could change names and it's used twice in src.name`

**Comment by Eveeifyeve** [comment]
But really I can't maintain it I just want to make it possible.

**Comment by Eveeifyeve** [comment]
Here if you want to continue on to it but I am moving this to an overlay.

```
{ lib
, stdenv
, rustPlatform
, fetchFromGitHub
, openssl
, pkg-config
, glibc
, libsoup
, cairo
, gtk3
, webkitgtk
, darwin
}:

let
  inherit (darwin.apple_sdk.frameworks) CoreServices Security SystemConfiguration;
in
rustPlatform.buildRustPackage rec {
  pname = "tauri";
  version = "2.0.0-beta.17";

  src = fetchFromGitHub {
    owner = "tauri-apps";
    repo = "tauri";
    rev = "tauri-${version}";
    hash = "sha256-M+L8R4DIiWBVYFSBYc6rcsxEqJipDfK0R5KqSTldQjE";
  };

  # Manually specify the sourceRoot since this crate depends on other crates in the workspace. Relevant info at
  # https://discourse.nixos.org/t/difficulty-using-buildrustpackage-with-a-src-containing-multiple-cargo-workspaces/10202
  sourceRoot = "${src.name}/tooling/cli";

  cargoHash = "sha256-45R8oyWwI1rgclhC7jehZQrDEPanJPqLSpg0v6TimQo=";

  nativeBuildInputs = [ 
	    pkg-config
	  ];
	
	  buildInputs = [ 
	    openssl
	  ] 
	  ++ lib.optionals stdenv.isLinux [ 
	    glibc libsoup cairo gtk3 webkitgtk 
	  ]
	  ++ lib.optionals stdenv.isDarwin [ 
	    CoreServices 
	    Security 
	    SystemConfiguration 
	  ];
	  
	  strictDeps = true;

  meta = {
	    description = "Build smaller, faster, and more secure desktop applications with a web frontend";
	    homepage = "https://tauri.app/";
	    license = with lib.licenses; [ asl20 /* or */ mit ];
	    mainProgram = "cargo-tauri-beta";
	    maintainers = with lib.maintainers; [ dit7ya happysalada eveeifyeve ];
	  };
	}
```

**Pull request closed without merging**